frontend/src/responseQuizView/responseQuizView.js

```diff
@@ -0,0 +1,343 @@
+    'run-on': 'There is a run on sentence',
+    'homophone': 'Misuse of a similar sounding word (homophone)',
+};
+
+export class ResponseQuizView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: null,
+            question: 1,
+            userAnswers: {},
+            score: 0,
+            graded: false,
+        };
+    }
+
+    async componentDidMount() {
+        try {
+            const response = await fetch(`/api/get_response_quiz_data/${this.props.textID}/`);
+            const data = await response.json();
+            this.setState({ data: data });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    nextQuestion() {
+        const nextQuestionNumber = this.state.question + 1;
+        if (nextQuestionNumber <= this.state.data.length) {
+            this.setState({
+                question: nextQuestionNumber,
+            });
+        } else {
+            console.log();
+        }
+    }
+
+    prevQuestion() {
+        const prevQuestionNumber = this.state.question - 1;
+        if (prevQuestionNumber > 0) {
+            this.setState({
+                question: prevQuestionNumber,
+            });
+        } else {
+            console.log();
+        }
+    }
+
+    getUnanswered() {
+        let numUnanswered = 0;
+        const answers = this.state.userAnswers;
+        for (let i = 1; i <= this.state.data.length; i++) {
+            if (answers[i] === undefined) {
+                numUnanswered++;
+            }
+        }
+        return numUnanswered;
+    }
+
+    gradeQuiz() {
+        let cSubmit;
+        if (this.getUnanswered() === 0) {
+            cSubmit = window.confirm('Are you sure that you want to submit your quiz? Your'
+                + ' answers are final.');
+        } else {
+            cSubmit = window.confirm('Are you sure that you want to submit your quiz? You'
+                + ' have ' + this.getUnanswered() + ' unanswered questions(s).');
+        }
+        if (cSubmit === true) {
+            this.setState({ graded: true });
+            let score = 0;
+            const answers = this.state.userAnswers;
+            for (let i = 0; i < this.state.data.length; i++) {
+                if (answers[i + 1] === this.state.data[i].answer) {
+                    score += 1;
+                }
+            }
+            this.setState({ score: score });
+        } else {
+            console.log();
+        }
+
+        // Brings the user back to the first question (beginning of the quiz) upon submission.
+        this.setState({ question: 1 });
+    }
+
+    onProgressBarClick = (event) => {
+        this.setState({
+            ...this.state,
+            question: parseInt(event.target.value),
+        });
+    }
+
+    onAnswerChoiceClick = (event) => {
+        const currentQ = this.state.question;
+        if ((typeof event.target.value) === 'undefined') {
```

Comment:
In which circumstances would the `value` here be `undefined`?

Reply:
If I remember correctly, for some reason, when answer choices are clicked, it would return an undefined event right next to an event with an actual value. I can check and see if this is still the case, though.

Reply:
This is still the case. I'm not entirely sure why, but an undefined event happens immediately before a substantive one. The current implementation is to make sure nothing breaks when this undefined event is caught.

However, the if-statement has since been simplified into one (rather than if...else if).